server/base/src/main/java/org/apache/accumulo/server/conf/ZooConfiguration.java

```diff
@@ -98,9 +102,26 @@ public String get(Property property) {
   }
 
   @Override
-  public boolean isPropertySet(Property prop) {
-    return fixedProps.containsKey(prop.getKey()) || getRaw(prop.getKey()) != null
-        || parent.isPropertySet(prop);
+  public boolean isPropertySet(Property prop, boolean cacheAndWatch) {
+    if (fixedProps.containsKey(prop.getKey())) {
+      return true;
+    }
+    if (cacheAndWatch) {
+      if (getRaw(prop.getKey()) != null) {
+        return true;
+      }
+    } else {
+      ZooReader zr = context.getZooReaderWriter();
+      String zPath = propPathPrefix + "/" + prop.getKey();
+      try {
+        if (zr.exists(zPath) && zr.getData(zPath, null) != null) {
+          return true;
+        }
+      } catch (KeeperException|InterruptedException e) {
+        throw new RuntimeException(e);
```

Comment:
I generally prefer a more specific RTE than the parent class. Maybe IllegalStateException or UncheckedIOException would be appropriate?